types/react-user-tour/index.d.ts

```diff
@@ -1,127 +1,161 @@
         /**
-        a react element representing the main body message of the tour step
-        */
+         * A react element representing the main body message of the tour step
+         */
         body: string | JSX.Element;
         /**
-        Optional properties horizontalOffset and verticalOffset values allow to move tooltip around pointed element.
-        */
+         * Optional properties horizontalOffset and verticalOffset values allow to move tooltip around pointed element.
+         */
         horizontalOffset?: number | undefined;
         /**
-        Optional properties horizontalOffset and verticalOffset values allow to move tooltip around pointed element.
-        */
+         * Optional properties horizontalOffset and verticalOffset values allow to move tooltip around pointed element.
+         */
         verticalOffset?: number | undefined;
         /**
-        Each step can also take an optional argument, position which will override the position of the tour component in relation to the selector that is determined by the application.
-        */
+         * Each step can also take an optional argument, position which will override the position of the tour component in relation to the selector that is determined by the application.
+         */
         position?: "left" | "right" | "top" | "topLeft" | "bottom" | "bottomLeft" | undefined;
     }
 
-    interface TourProps extends HTMLAttributes<any> {
+    interface TourProps {
         /**
-        A boolean value representing whether or not the tour should currently be displayed
-        */
+         * A boolean value representing whether or not the tour should currently be displayed
+         */
         active: boolean;
         /**
-        An integer representing the current active step of the tour
-        */
+         * An integer representing the current active step of the tour
+         */
         step: number;
         /**
-        function that fires when user clicks the Next button.
-        Receives the next step integer as a callback.
-        For example, if current step is 1 and user clicks the Next button, onNext(2) will be called.
-        */
-        onNext: Function;
-        /**
-        function that fires when user clicks the Back button.
-        Receives the previous step integer as a callback.
-        For example, if current step is 2 and user clicks the Back button, onBack(1) will be called.
-        */
-        onBack: Function;
-        /**
-        function that fires when user clicks the X button or the Done Button.
-        */
-        onCancel: Function;
-        /**
-        An array of TourStep.
-        */
+         * Function that fires when user clicks the Next button.
+         * Receives the next step integer as a callback.
+         * For example, if current step is 1 and user clicks the Next button, onNext(2) will be called.
+         *
+         * @default () => {}
+         */
+        onNext: (step: number) => void;
```

Comment:
It's odd that these callbacks say they have defaults, but none of them are optional. Seems wrong?

Reply:
Yes you are right. I have made them optional in latest repush.